### PR TITLE
DRAFT: [CIT-623] Test reporting for system tests

### PR DESCRIPTION
**Pull request by abelikt**
## Proposed changes

Add reporting for system tests as a separate workflow.

- Open Issue: Unsure if this makes sense at another ci repo.
- Open issue: If no system test result was found, we traverse back to the last success state. With this we cannot guarantee, that the system-test runs belong togehter.
- Open issue: not yet suited for the official repo as there are different runner workflows.

- TODO: Add reporting to official system tests
- TODO: DEAL-Breaker: Postprocessing of skipped tests does makes them appear to be green
- TODO: Handling of Workflows that failed completely (Old ones will be put into the report)
- TODO: Add overall report-info : Log where do the reports come from etc 
- TODO: Move contents to github pages
- TODO: Improve build script to make it more understandable
- TODO: Run regulary and find out how we do this accros repos

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

https://github.com/thin-edge/thin-edge.io/issues/623

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by abelikt** [comment]
See report matrix here as an example:
https://github.com/abelikt/thin-edge.io/actions/runs/1545725403

File: .github/workflows/generate-report.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    steps:
+
+    - name: checkout
+      uses: actions/checkout@v2
+
+    - name: Generate Report
+      run:  bash ./ci/report/build.sh
+      continue-on-error: false
+      env:
+        THEGHTOKEN: ${{secrets.GITHUB_TOKEN}}
```
Comment (makr11st) [inline .github/workflows/generate-report.yml:23]:
```suggestion
        THEGHTOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Review by makr11st — COMMENTED** [review]
It's looks a little bit like dark magic, but one thing I like about it is that most of the logic is done in scripts and they are just called by the CI which makes them much easier to run locally than other workflows we have.

**Comment by abelikt** [comment]
The content of this PR was updated and got merged with the new ci-pipeline.

**Pull request closed without merging**